libssh2-tunnel-example.c

```diff
@@ -106,7 +106,10 @@ int forward_tunnel(LIBSSH2_SESSION *session, LIBSSH2_CHANNEL *channel) {
             }
             wr = 0;
             do {
-                i = libssh2_channel_write(channel, buf, len);
+                i = LIBSSH2_ERROR_EAGAIN;
+                while (i == LIBSSH2_ERROR_EAGAIN) {
```

Comment:
This can cause an infinite loop, blocking the tunnel on write.
The correct way to handle this is to check if the socket is alive before sending it again: https://stackoverflow.com/a/9811061/2052076

Some code examples that I found:
https://github.com/allisrc/libgit2-allisrc/blob/62ab0e1fc2a471db960b0bd881aab116b350a961/src/transports/ssh.c#L266

https://github.com/u35s/xtools/blob/99fe76fffb5d893573b1aa21e8032dc5385c2837/src/common/ssh2_client.cc#L185